### ISSUE TITLE
Why does BackgroundExecutingUseCase expect client to give Dispatcher.Default?

### ISSUE THREAD

**Issue by rhmndnt** [MrHadiSatrio/Journal3#1]
https://github.com/MrHadiSatrio/Journal3/blob/30ff779807a5e171d9b86fb0d8e463836fa27039/app-android-journal3/src/main/kotlin/com/hadisatrio/apps/android/journal3/story/ViewStoriesActivity.kt#L48

This decorator's name suggests something like this:
```
class BackgroundExecutingUseCase(
    private val coroutineScope: CoroutineScope,
    private val origin: UseCase
) : UseCase {

    override fun invoke() {
        coroutineScope.launch(Dispatchers.Default) { origin() }
    }
}
```

**Comment by MrHadiSatrio** [comment]
This is actually a valid concern; thanks for raising it up!

In my opinion, there are two ways this can be fixed. Either we keep the name and abstract the dispatcher used (like your example), or we rename it to something like `CoroutineDispatchingUseCase` and keep expecting the client to provide a `Dispatcher`.

I'm leaning more towards the second, as there might be a case where a particular `UseCase` require a non-default dispatcher.

What do you think?

**Comment by rhmndnt** [comment]
`CoroutineDispatchingUseCase` would be great

**Comment by MrHadiSatrio** [comment]
This has been resolved with 4a0389c82510339a12ed8b64f4cb23268063b4f5; we now have `CoroutineDispatchingUseCase` which accepts `CoroutineDispatcher` as an argument.
